### PR TITLE
add autoconf 2.69 to the image

### LINKED ISSUE #71: autoconf version
The version of autoconf shipping with CentOS 5 is sufficiently out of date to prevent building various projects (among them libffi). It's easy to install it yourself, but since this seems likely to be a common problem for users of the manylinux1 image I'd be happy to send a PR to install the latest autoconf (2.69, released 4 years ago) as part of the build process if people think it belongs here.

(It really just looks like `RUN wget http://ftp.gnu.org/gnu/autoconf/autoconf-latest.tar.gz && tar zxvf autoconf-latest.tar.gz && cd autoconf* && ./configure && make install`)

### PR DESCRIPTION
**Pull request by reaperhulk**
Fixes #71

### CODE REVIEW

File: docker/build_scripts/build_utils.sh [hunk 1]
```diff
@@ -140,3 +142,24 @@ function build_curl {
     (cd ${curl_fname} && do_curl_build)
     rm -rf ${curl_fname} ${curl_fname}.tar.bz2
 }
+
+
+function do_autoconf_build {
```
Comment (matthew-brett) [inline docker/build_scripts/build_utils.sh]: Maybe rename as `do_standard_install` or something?

**Comment by matthew-brett** [comment]
Great - thanks for the neat work.

File: docker/build_scripts/build_utils.sh (hunk 1 above)
Comment (reaperhulk) [inline docker/build_scripts/build_utils.sh]: Done!

File: docker/build_scripts/build_utils.sh [hunk 2]
```diff
@@ -140,3 +142,24 @@ function build_curl {
...
+    make install > /dev/null
+}
+
+
+function build_autoconf {
+    local autoconf_fname=$1
+    check_var ${autoconf_fname}
+    local autoconf_sha256=$2
+    check_var ${autoconf_sha256}
+    check_var ${CURL_DOWNLOAD_URL}
```
Comment (matthew-brett) [inline docker/build_scripts/build_utils.sh]: Did you mean to check the curl download URL here?

**Comment by reaperhulk** [comment]
One outstanding question -- do we want to remove the `autoconf` from the yum install? Chances are something else depends on it so it'll be implicitly pulled in during dependency resolution thus making the change semi-pointless, but removing it shouldn't hurt.

**Comment by matthew-brett** [comment]
I guess you could move the autoconf build up to before the Python build steps, just after the yum install, and remove the autoconf install, and see what happens.

**Comment by reaperhulk** [comment]
Moving the autoconf to before the yum install shouldn't really change anything since the RPMs will declare dependencies on the autoconf RPM and won't be looking to see if the binary is available or not. I went ahead and looked in the image to see what depends on autoconf:

```
[root@22e5a419f7ea /]# rpm -q --whatrequires autoconf
automake-1.9.6-3.el5
```

So looks like there's no real point to removing it from the deps list, as automake will pull it right back in.

File: docker/build_scripts/build_utils.sh (hunk 2 above)
Comment (reaperhulk) [inline docker/build_scripts/build_utils.sh]: Sure did, fixed.

**Comment by matthew-brett** [comment]
Sure, I was suggesting moving it up _after_ the yum install before the Python build steps.  I think it's neater to remove autoconf from stated dependencies even if in practice it still gets installed.

**Comment by reaperhulk** [comment]
Ah, I understand. I'll push a commit shortly that does that :)

**Comment by matthew-brett** [comment]
Great - thanks - merging.

**Comment by njsmith** [comment]
Unfortunately due to Travis issues, as of right now this change still hasn't landed in the official images -- see #76

**Pull request merged**